src/createMessage.ts

```diff
@@ -69,7 +70,7 @@ export async function createApplicationMessage(
   message: Uint8Array,
   cs: CiphersuiteImpl,
   authenticatedData: Uint8Array = new Uint8Array(),
-) {
+): Promise<{ newState: ClientState; privateMessage: PrivateMessage }> {
```

Comment:
This is needed because typescript wasn't able to infer a named type from the return